tests/suites/casper/content.js

```diff
@@ -2,19 +2,31 @@
 var fs = require("fs");
 
 casper.test.begin("Casper.getPageContent() text/html content", 1, function(test) {
-  casper.start("tests/site/test.html", function() {
-    test.assertMatch(this.getPageContent(), /<title>CasperJS test target/,
-                      "Casper.getPageContent() retrieves text/html content");
-  }).run(function() {
-    test.done();
-  });
+    casper.start("tests/site/test.html");
+    casper.waitForUrl("tests/site/test.html", function() {
+        test.assertMatch(this.getPageContent(), /<title>CasperJS test target/,
+                         "Casper.getPageContent() retrieves text/html content");
+    });
+    casper.run(function() {
+        test.done();
+    });
 });
 
-casper.test.begin("Casper.getPageContent() non text/html content", 1, function(test) {
-  casper.start("tests/site/dummy.js", function() {
-    test.assertEquals(this.getPageContent(), "document.write('foo');\n",
-                      "Casper.getPageContent() retrieves non text/html content");
-  }).run(function() {
-    test.done();
-  });
+casper.test.begin("Casper.getPageContent() non text/html content", 2, function(test) {
+    // NOTE due to a bug in slimerjs we can only use text/* and
+    // application/json content here
```

Comment:
Ok, perhaps we should open an issue. Would you @istr be willing to do that, as you have the best current knowledge about the subject?